File: sdk/swift/Sources/HiTechCloudSDK/Ssl.swift
import Foundation

/// SSL Certificates: list, order, download certificates
public class SSLResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/certificate - List all SSL certificates
    public func list(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/certificate", params: params)
    }

    /// GET /api/certificate/{id} - Return certificate details
    public func get(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/certificate/{id}", params: params)
    }

    /// GET /api/certificate/{id}/crt - Download X.509 certificate
    public func download(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/certificate/{id}/crt", params: params)
    }

    /// GET /api/certificate/order - List available certificates for purchase
    public func listAvailable(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/certificate/order", params: params)
    }

    /// POST /api/certificate/order - Order new certificate
    public func order(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/certificate/order", params: params)
    }

    /// GET /api/certificate/order/{product_id}/software - List server software
    public func listServerSoftware(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/certificate/order/{product_id}/software", params: params)
    }

}

File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Ssl.kt
package vn.hitechcloud.sdk

/** SSL Certificates: list, order, download certificates */
class SSLResource(private val httpClient: HttpClient) {
    /** GET /api/certificate - List all SSL certificates */
    fun list(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/certificate", params)
    }

    /** GET /api/certificate/{id} - Return certificate details */
    fun get(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/certificate/{id}", params)
    }

    /** GET /api/certificate/{id}/crt - Download X.509 certificate */
    fun download(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/certificate/{id}/crt", params)
    }

    /** GET /api/certificate/order - List available certificates for purchase */
    fun listAvailable(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/certificate/order", params)
    }

    /** POST /api/certificate/order - Order new certificate */
    fun order(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/certificate/order", params)
    }

    /** GET /api/certificate/order/{product_id}/software - List server software */
    fun listServerSoftware(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/certificate/order/{product_id}/software", params)
    }

}
